### PR TITLE
Extend debug message for syncToRealTime() function

### PR DESCRIPTION
**Pull request by uturuncoglu**
This PR aims to extend error messages of syncToRealTime() function for the calendars that does not support leap year. The message is also specialized for based on used calendar. For testing, `libfaketime` library is used by running test application using `srun --label -n 4 /work/noaa/nems/tufuk/ESMF/libfaketime/bin/faketime '2024-02-29 08:00:00' ./esmApp` command.

Linked issue: Fixes esmf-org/esmf-support#423

### CODE REVIEW

**Comment by uturuncoglu** [comment]
@oehmke I modified the underlying function to throw error for the cases that do not support leap year. I also split the error message and make it specific for the used calendar to be more clear. I am currently performing one more set of test to be sure it is working as expected. Please let me know if you want any change. Thanks.

**Comment by uturuncoglu** [comment]
@oehmke BTW, are we expecting to have successful run from this function for the leap years but the dates except Feb 29th. In this case, independent from the clock it is just looking to the calendar. Anyway, let me know what you think.

File: src/Infrastructure/TimeMgr/src/ESMCI_Time.C [hunk 1]
```diff
@@ -830,13 +830,48 @@ namespace ESMCI{
...
+            "; calkindflag is NOLEAP. Only GREGORIAN and JULIAN are supported.",
+            ESMC_CONTEXT, &rc);
+          return(rc);
+          break;
+
+        case ESMC_CALKIND_360DAY:
+          ESMC_LogDefault.MsgFoundError(ESMC_RC_ARG_WRONG,
+            "; calkindflag is 360DAY. Only GREGORIAN and JULIAN are supported.",
+            ESMC_CONTEXT, &rc);
+          break;
```
Comment (oehmke) [inline src/Infrastructure/TimeMgr/src/ESMCI_Time.C:846]: Hi Ufuk, I noticed that there wasn't a return here and I think that there should be, would you please confirm that. Thanks!

File: src/Infrastructure/TimeMgr/src/ESMCI_Time.C [hunk 2]
```diff
@@ -830,13 +830,48 @@ namespace ESMCI{
...
-        this->calendar->calkindflag == ESMC_CALKIND_NOCALENDAR) {
-      ESMC_LogDefault.MsgFoundError(ESMC_RC_ARG_WRONG,
-        "; calkindflag is JULIANDAY, "
-        "MODJULIANDAY or NOCALENDAR.", ESMC_CONTEXT, &rc);
-      return(rc);
+    switch (this->calendar->calkindflag)
+    {
+        case ESMC_CALKIND_NOLEAP:
+          ESMC_LogDefault.MsgFoundError(ESMC_RC_ARG_WRONG,
+            "; calkindflag is NOLEAP. Only GREGORIAN and JULIAN are supported.",
```
Comment (oehmke) [inline src/Infrastructure/TimeMgr/src/ESMCI_Time.C:837]: What do you think about adding "when syncing to real time." to the end of the error message?  Just to make it more clear?

File: src/Infrastructure/TimeMgr/src/ESMCI_Time.C (hunk 1 above)
Comment (uturuncoglu) [reply src/Infrastructure/TimeMgr/src/ESMCI_Time.C:847]: @oehmke Good catch. Thanks. I add it.

File: src/Infrastructure/TimeMgr/src/ESMCI_Time.C (hunk 2 above)
Comment (uturuncoglu) [reply src/Infrastructure/TimeMgr/src/ESMCI_Time.C]: @oehmke Okay. I add it.

**Comment by oehmke** [comment]
@uturuncoglu, looks great to me! Feel free to merge whenever you like.

**Comment by oehmke** [comment]
@uturuncoglu, sorry I accidentally closed it, but it's reopen now. Merge whenever you'd like.

**Comment by uturuncoglu** [comment]
@oehmke No worries. I have just merge it.

**Comment by oehmke** [comment]
Great! Thanks.

**Pull request merged**